dashboard/src/shared/url.ts

```diff
@@ -97,7 +97,7 @@ export const backend = {
     update: (cluster: string, namespace: string, name: string) =>
       `${backend.apprepositories.base(cluster, namespace)}/${name}`,
   },
-  canI: (cluster: string) => `api/v1/clusters/${cluster}/can-i`,
+  canI: (cluster: string) => `api/v1/clusters${cluster ? "/" + cluster : ""}/can-i`,
```

Comment:
Hmm, depends what you're doing in the case where `cluster` isn't defined, but sounds like it could be dangerous (EDIT: see below). I'm guessing this is because some code in the frontend is checking whether users can do things with app repositories which are on the kubeapps cluster?

Reply:
Well, the current behavior is: 
1. performs a POST request to `api/v1/clusters//can-i`
2. got a 302/3 redirect (don't remember) to `api/v1/clusters/can-i`
3. browser (ignoring the RFC) changes POST to GET methdod
4. kubeops receives GET `api/v1/clusters/can-i`, returning a 405 method not allowed

I think we should still handle this route for avoiding odd browser behavior, don't you think?
Regarding the can-i logic, replying separately.

Reply:
> I think we should still handle this route for avoiding odd browser behavior, don't you think? I mean, I'd rather go for consistent error codes instead of having these redirections regardless of the implemented logic.

What are the call-sites that are calling this without a cluster name? I would think that's the issue - the dashboard should not be trying to check if the user has perms without a cluster name. But that might need some investigation (ie. separate PR).

Reply:
Ok, I totally agree, let's remove all can-i stuff here and let me file another issue for the future. We better have this PR more focused, you're right